### PR TITLE
fix(PetitioResponse): support non-ascii json responses

### LINKED ISSUE #10: Non-alphabetical texts like korean get crashed
```diff
# original text:
- 타락한 그림자의 반지
# when petitio parses it:
+ í\x83\x80ë\x9D½í\x95\x9C ê·¸ë¦¼ì\x9E\x90ì\x9D\x98 ë°\x98ì§\x80
```

Using petitio v1.1.0

When I request to [this url](https://ludy.game.onstove.com:44333/web/ItemDictionary/Select/212300022?_=1617310085179), petitio is returning something like:

```json
{
    "ItemInfo": {
        "AcquisitionInfo": {
            "Acquisition_000": [
                null
            ]
        },
        "BasicInfo": {
            "Tooltip_Item_000": [
                null
            ],
            "categoryName": "<FONT COLOR='#FA5D00'>ì ë¬¼ ë°</FONT>",
            "categoryType": "200",
            "enhance": "false",
            "itemId": "212300022",
            "itemLevel": "ì
                           ì
 ë ë²¨ 960",
            "itemName": "íë ê·¸ë¦¼ì",
            "maxQualityIndex": "0",
            "minQualityIndex": "0",
            "setItem": "false"
        }
    }
}
```

but the expected output was:

```json
{
    "ItemInfo": {
        "AcquisitionInfo": {
            "Acquisition_000": [
                null
            ]
        },
        "BasicInfo": {
            "Tooltip_Item_000": [
                null
            ],
            "categoryName": "<FONT COLOR='#FA5D00'>유물 반지</FONT>",
            "categoryType": "200",
            "enhance": "false",
            "itemId": "212300022",
            "itemLevel": "아이템 레벨 960",
            "itemName": "타락한 그림자의 반지",
            "maxQualityIndex": "0",
            "minQualityIndex": "0",
            "setItem": "false"
        }
    }
}
```

It seems while parsing buffer to json, something is causing to crash non-alphabetical letters like korean.

Code to reproduce:
```ts
import fetch from "petitio"

fetch("https://ludy.game.onstove.com:44333/web/ItemDictionary/Select/212300022?_=1617310085179").json().then(console.log)
```

I'm using nodejs v14.16.0 with linux mint 20.1.

### PR DESCRIPTION
**Pull request by zero734kr**
Closes #10 

Using Buffer#toString fixed this because it parses string to utf-8 automatically.

### CODE REVIEW

**Comment by zero734kr** [comment]
Ah, no problem if you prefer to delete the ``tests/checkKorean.test.ts`` file because I used it only to test if it was really working

**Review by tbnritzdoge — CHANGES_REQUESTED** [review]
Once the Korean Tests are removed this looks good to merge although we may want to incorporate a encoding type in buffer#toString like in PetitioResponse#text. I will invoke @Nytelife26 to see his opinion on that. Thanks for the PR :)

**Review by tbnritzdoge — CHANGES_REQUESTED** [review]
Once the Korean Tests are removed this looks good to merge although we may want to incorporate a encoding type in buffer#toString like in PetitioResponse#text. I will invoke @Nytelife26 to see his opinion on that. Thanks for the PR :)

**Review by tbnritzdoge — CHANGES_REQUESTED** [review]
Once the Korean Tests are removed this looks good to merge although we may want to incorporate a encoding type in buffer#toString like in PetitioResponse#text. I will invoke @Nytelife26 to see his opinion on that. Thanks for the PR :)

**Comment by tbnritzdoge** [comment]
Apologies for spamming this, my internet was having issues.

**Comment by zero734kr** [comment]
@tbnritzdoge applied edits for change requests as you requested 👍

File: src/lib/PetitioResponse.ts [hunk 1]
```diff
@@ -56,10 +56,11 @@ export class PetitioResponse {
 
 	/**
 	 * @template T Type casting parameter for the JSON result.
+         * @param {BufferEncoding} [encoding="utf8"] The encoding to use before parsing to json.
```
Comment (tbnritzdoge) [inline src/lib/PetitioResponse.ts]: formatting

**Review by tbnritzdoge — CHANGES_REQUESTED** [review]

**Comment by lveneris** [comment]
The reason we didn't use `toString` or the like it is because it is approximately 3x slower according to our testing. I will look into ways to rectify this now, perhaps if there is a way to use `fromCharCode` with string encoding. Thank you for the PR!

File: src/lib/PetitioResponse.ts [hunk 2]
```diff
@@ -56,10 +56,11 @@ export class PetitioResponse {
 
 	/**
 	 * @template T Type casting parameter for the JSON result.
+         * @param {BufferEncoding} [encoding="utf8"] The encoding to use before formatting to json.
```
Comment (Nytelife26) [inline src/lib/PetitioResponse.ts:59]: Why was the whitespace on this line changed?

**Comment by lveneris** [comment]
Nevermind, looks good to me. Ready to merge once CI passes.

**Pull request merged**